### PR TITLE
Remove Templating Helper From Other Helpers & Subscribers

### PR DESCRIPTION
**Pull request by mabumusa1**
**Waiting for #12124 to be merged**

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

**Comment by mabumusa1** [comment]
Helpers that need work 

- app/bundles/CoreBundle/Helper/ThemeHelper.php
- app/bundles/EmailBundle/Helper/MailHelper.php

File: app/config/config_test.php [hunk 1]
```diff
@@ -30,7 +30,8 @@
 $container->loadFromExtension('framework', [
     'test'    => true,
     'session' => [
-        'storage_id' => 'session.storage.filesystem',
+        'storage_id' => 'session.storage.mock_file',
```
Comment (mabumusa1) [inline app/config/config_test.php:34]: @escopecz this change is based on the [docs here](https://github.com/liip/LiipFunctionalTestBundle/blob/master/doc/installation.md)

File: app/bundles/CoreBundle/Test/AbstractMauticTestCase.php [hunk 2]
```diff
@@ -105,7 +105,7 @@ protected function loadFixtureFiles(array $paths = [], bool $append = true): arr
 
     private function mockServices(): void
     {
-        self::$container->set('session', new Session(new FixedMockFileSessionStorage()));
+        //self::$container->set('session', new Session(new FixedMockFileSessionStorage()));
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Test/AbstractMauticTestCase.php]: @escopecz this line is not needed anymore, since `LiipFunctionalTestBundle` starts its own sessions, this line will break sessions. if you agree on the removal of this method I will proceed

File: app/bundles/ApiBundle/Tests/Functional/Oauth2Test.php [hunk 3]
```diff
@@ -79,7 +79,7 @@ public function testAuthWithInvalidAccessToken(): void
 
     public function testAuthWorkflow(): void
     {
-        $this->client->enableReboot();
+        $this->client->disableReboot();
```
Comment (mabumusa1) [inline app/bundles/ApiBundle/Tests/Functional/Oauth2Test.php:81]: @escopecz I adopted this solution because this test failed, This is the [source](https://stackoverflow.com/questions/62328408/cannot-set-session-id-after-the-session-has-started-at-mockarraysessionstorage-p)

Comment (escopecz) [reply app/bundles/ApiBundle/Tests/Functional/Oauth2Test.php:81]: It's disabled by default in `MauticMysqlTestCase` so this could be removed, but it's not necessary.

File: app/bundles/CoreBundle/Test/AbstractMauticTestCase.php (hunk 2 above)
Comment (escopecz) [reply app/bundles/CoreBundle/Test/AbstractMauticTestCase.php]: Let's remove these "mock services" completely. It was breaking my test last year so I refactored it to be toggleable. If we can remove it and all tests are passing then let's do that.

File: app/bundles/FormBundle/Model/SubmissionModel.php [hunk 4]
```diff
@@ -585,15 +585,15 @@ function () use ($results, $form, $translator, $viewOnlyFields) {
 
                 return $response;
             case 'html':
-                $content = $this->templatingHelper->getTemplating()->renderResponse(
+                $content = $this->templatingHelper->getTemplating()->render(
                     'MauticFormBundle:Result:export.html.twig',
```
Comment (mabumusa1) [inline app/bundles/FormBundle/Model/SubmissionModel.php:589]: @escopecz I am leaving this here on purpose, it will be changed in another PR

File: app/bundles/FormBundle/Model/SubmissionModel.php [hunk 5]
```diff
@@ -739,14 +739,14 @@ function () use ($results, $translator) {
 
                 return $response;
             case 'html':
-                $content = $this->templatingHelper->getTemplating()->renderResponse(
+                $content = $this->templatingHelper->getTemplating()->render(
```
Comment (mabumusa1) [inline app/bundles/FormBundle/Model/SubmissionModel.php:742]: @escopecz I am leaving this for another PR

File: app/bundles/PluginBundle/Helper/IntegrationHelper.php [hunk 6]
```diff
@@ -560,7 +559,7 @@ public function getShareButtons()
 
                 //add the api keys for use within the share buttons
                 $shareSettings['keys']   = $apiKeys;
-                $shareBtns[$integration] = $templating->render($plugin->getBundle().":Integration/$integration:share.html.twig", [
+                $shareBtns[$integration] = $this->twig->render($plugin->getBundle().":Integration/$integration:share.html.twig", [
```
Comment (mabumusa1) [inline app/bundles/PluginBundle/Helper/IntegrationHelper.php]: Note to self, we might find places where regex did not detect the change like `":Integration/$integration:share.html.twig"`

File: app/bundles/CoreBundle/Test/AbstractMauticTestCase.php (hunk 2 above)
Comment (mabumusa1) [reply app/bundles/CoreBundle/Test/AbstractMauticTestCase.php]: @escopecz I just removed it completely, tests are passing

**Review by escopecz — APPROVED** [review]
Looks good. I noticed a 500 error unrelated to these changes during testing the global search and fixed it with https://github.com/mautic/mautic/pull/12122/commits/ab60fd038709abfec7bc50f805156dd5a8c8a668
